### PR TITLE
endpoint: use async snapshot

### PR DESCRIPTION
**Pull request by overvenus**
## What have you changed? (mandatory)

This reverts commit ec53f365c07e78c086c9a73e216acce4c2c5c918.

## What are the type of the changes? (mandatory)

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

unit test.

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No

## Refer to a related PR or issue link (optional)

https://github.com/tikv/tikv/pull/2148

## Benchmark result if necessary (optional)

From 85872.5 to 86972.7. I think the different is within the margin of error. The bottleneck is endpoint worker, it has reached 100% CPU time.

### CODE REVIEW

**Comment by breeswish** [comment]
Does it mean that we can totally use read pool for coprocessor now? (currently the process of retrieving snapshot is not running on read pool)

File: src/coprocessor/metrics.rs [hunk 1]
```diff
@@ -11,7 +11,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use prometheus::*;
+use prometheus::{exponential_buckets, HistogramVec, IntCounterVec, IntGaugeVec};
```
Comment (breeswish) [inline src/coprocessor/metrics.rs]: BTW I prefer to use * here. It is very unlikely to have collisions for these names.

**Comment by siddontang** [comment]
do we need to remove the group from scheduler too?

**Pull request merged**